### PR TITLE
[Tuning] Potential Masquerading as System32 DLL

### PR DESCRIPTION
**Pull request by Samirbous**
excluding top noisy patterns.

### CODE REVIEW

**Review by Aegrah — APPROVED** [review]

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml [hunk 1]
```diff
@@ -40,9 +40,24 @@ library where event.action == "load" and dll.Ext.relative_file_creation_time <=
         "?:\\$WINDOWS.~BT\\Work\\*",
         "?:\\Windows\\WinSxS\\*",
         "?:\\Windows\\SoftwareDistribution\\Download\\*",
-        "?:\\Windows\\assembly\\NativeImages_v*"
+        "?:\\Windows\\assembly\\NativeImages_v*", 
+        "E:\\docker\\windowsfilter\\*", 
+        "E:\\System\\Users\\*",
+        "\\\\?\\Volume{*}\\Windows\\*"
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_masquerading_windows_dll.toml:46]:
```suggestion
        "?:\\Windows\\assembly\\NativeImages_v*"
```
Those seem to be env-specific, each of them only occurs in one cluster

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml [hunk 2]
```diff
@@ -40,9 +40,24 @@ library where event.action == "load" and dll.Ext.relative_file_creation_time <=
...
       )
     ) and
+    not process.executable : 
+              ("C:\\Packages\\Plugins\\Microsoft.GuestConfiguration.ConfigurationforWindows\\*\\dsc\\GC\\gc_*.exe",
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_masquerading_windows_dll.toml:50]:
```suggestion
    not process.executable : (
```
This one seems to be limited to one env too

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml [hunk 3]
```diff
@@ -40,9 +40,24 @@ library where event.action == "load" and dll.Ext.relative_file_creation_time <=
...
+        "?:\\Windows\\assembly\\NativeImages_v*", 
+        "E:\\docker\\windowsfilter\\*", 
+        "E:\\System\\Users\\*",
+        "\\\\?\\Volume{*}\\Windows\\*"
       )
     ) and
+    not process.executable : 
+              ("C:\\Packages\\Plugins\\Microsoft.GuestConfiguration.ConfigurationforWindows\\*\\dsc\\GC\\gc_*.exe", 
+               "?:\\Program Files\\*\\*.exe", 
+               "?:\\Program Files (x86)\\*\\*.exe",
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_masquerading_windows_dll.toml:52]:
```suggestion
```
I think we should avoid those big exceptions, as this is a BBR, we can afford some FPs

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml [hunk 4]
```diff
@@ -54,7 +69,7 @@ library where event.action == "load" and dll.Ext.relative_file_creation_time <=
...
...n.dll", "sortwindows63.dll", "sstpcfg.dll", "tdhres.dll", "umpodev.dll", "utcapi.dll", "windlp.dll", "wow64base.dll", "wow64con.dll", "blbuires.dll", "bpainst.dll", "cbclient.dll", "certadm.dll", "certocm.dll", "certpick.dll", "csdeployres.dll", "dsdeployres.dll", "eapa3hst.dll", "eapacfg.dll", "eapahost.dll", "elsext.dll", "encdump.dll", "escmigplugin.dll", "fsclient.dll", "fsdeployres.dll", "fssminst.dll", "fssmres.dll", "fssprov.dll", "ipamapi.dll", "kpssvc.dll", "lbfoadminlib.dll", "mintdh.dll", "mmci.dll", "mmcico.dll", "mprsnap.dll", "mstsmhst.dll", "mstsmmc.dll", "muxinst.dll", "personax.dll", "rassfm.dll", "rasuser.dll", "rdmsinst.dll", "rdmsres.dll", "rtrfiltr.dll", "sacsvr.dll", "scrdenrl.dll", "sdclient.dll", "sharedstartmodel.dll", "smsrouter.dll", "spwizimg_svr.dll", "sqlcecompact40.dll", "sqlceoledb40.dll", "sqlceqp40.dll", "sqlcese40.dll", "srvmgrinst.dll", "svrmgrnc.dll", "tapisnap.dll", "tlsbrand.dll", "tsec.dll", "tsprop.dll", "tspubiconhelper.dll", "tssdjet.dll", "tsuserex.dll", "ualapi.dll", "ualsvc.dll", "umcres.dll", "updatehandlers.dll", "usocore.dll", "vssui.dll", "wsbappres.dll", "wsbonline.dll", "wsmselpl.dll", "wsmselrr.dll", "xpsfilt.dll", "xpsshhdr.dll"
```
Comment (w0rk3r) [inline rules_building_block/defense_evasion_masquerading_windows_dll.toml:72]: `"authz.dll"` & `"mfc42.dll", "mfc42u.dll",`

Had matches in one environment only, maybe we should include them back

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml (hunk 1 above)
Comment (Samirbous) [reply rules_building_block/defense_evasion_masquerading_windows_dll.toml:46]: `\\\\?\\Volume{*}\\Windows\\` is not env specific, we can limit to `C:\` drive too but we will miss executed from mounted ISO/VHD files.

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml (hunk 3 above)
Comment (Samirbous) [reply rules_building_block/defense_evasion_masquerading_windows_dll.toml:52]: avoiding those big exception generate lot of FPs, its not some FPs. this risk to miss true TPs due to high volume, also programfiles is writable by local admin (which limit the abuse scope).

File: rules_building_block/defense_evasion_masquerading_windows_dll.toml (hunk 4 above)
Comment (Samirbous) [reply rules_building_block/defense_evasion_masquerading_windows_dll.toml:72]: only one env we see, we should expect to have similar FPs in other env. if you want to go granular you should tune it by dll.hash, which is not easy to maintain. goal is to have a balance (coverage, noise, tuning efforts).

**Pull request closed without merging**